### PR TITLE
Make sure FindLAPACK finds the same BLAS library

### PR DESCRIPTION
**Pull request by isuruf**
BLAS library is found by cmake/Dependencies.cmake and then
LAPACK library is found by FindLAPACK.cmake which in turn calls
FindBLAS.cmake. This means that we are searching for BLAS twice
and they might be different things. By setting a few variables,
this can be avoided.

cc @seemethere

### CODE REVIEW

**Comment by isuruf** [comment]
It would be good to merge this for the conda-forge build.

cc @rgommers

**Review by rgommers — APPROVED** [review]
Thanks @isuruf, this looks good to me. 

@malfet would you be able to land this?

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -152,6 +155,9 @@ elseif(BLAS STREQUAL "MKL")
     include_directories(AFTER SYSTEM ${MKL_INCLUDE_DIR})
     list(APPEND Caffe2_PUBLIC_DEPENDENCY_LIBS caffe2::mkl)
     set(CAFFE2_USE_MKL ON)
+    set(BLAS_INFO "mkl")
+    set(BLAS_FOUND 1)
+    set(BLAS_LIBRARIES ${MKL_LIBRARIES})
```
Comment (malfet) [inline cmake/Dependencies.cmake:160]: For consistency with statement on line 153, shouldn't it be `caffe2::mkl`?

**Review by malfet — APPROVED** [review]
LGTM, although it misses similar checks for Atlas and Eigen.

**Review by seemethere — APPROVED** [review]

**Comment by isuruf** [comment]
> LGTM, although it misses similar checks for Atlas and Eigen.

I added support atlas. Eigen is not supported by `cmake/Modules/FindBLAS.cmake`

**Comment by rgommers** [comment]
This seems ready to land. The three CI failures are all unrelated (1 a connectivity issue, 2 timeouts)

**Pull request closed without merging**